app/jobs/benchmark_regression_job.rb

```diff
@@ -0,0 +1,72 @@
+require 'uri'
+require 'net/http'
+require 'json'
+
+class BenchmarkRegressionJob < ActiveJob::Base
+  queue_as :default
+
+  @@highest = Hash.new(10000) # any max value
+  @@flag = Hash.new(false)
+
+  def perform(benchmark_run_id)
+
+    benchmark_run = BenchmarkRun.find_by_id(benchmark_run_id)
+
+    categories = benchmark_run.result.keys
+
+    current_commit = benchmark_run.initiator
+    previous_commits = Commit.where('created_at < ?', current_commit.created_at).where(repo: current_commit.repo).limit(1000)
+
+    previous_benchmark_runs = []
+    previous_commits.each do |commit|
+      previous_benchmark_runs << BenchmarkRun.find_by(initiator: commit, benchmark_type: benchmark_run.benchmark_type,
+                       benchmark_result_type: benchmark_run.benchmark_result_type)
+    end
+
+    categories.each do |category|
+      previous_benchmark_results = previous_benchmark_runs.map { |run| run.result[category].to_f }
+
+      results_param = average(previous_benchmark_results) + 2*standard_deviation(previous_benchmark_results)
+      results_upper_cutoff = @@highest[category] - standard_deviation(previous_benchmark_results)
+
+      if benchmark_run.result[category].to_f > results_param && benchmark_run.result[category].to_f <= results_upper_cutoff
+        create_issue benchmark_run, category, results_param
+        @@highest[category] = benchmark_run.result[category].to_f
+        @@flag[category] = true
+      end
+      if @@flag[category] == true && benchmark_run.result[category].to_f < results_param
+        @@highest[category] = 10000
+        @@flag = false
+      end
```

Comment:
Ask me if you need any detailed explanation. Not sure if the task can be done this way, but it will satisfy our conditions.